projects/plugins/jetpack/extensions/shared/external-media/sources/google-photos/index.js

```diff
@@ -15,23 +16,50 @@ function GooglePhotos( props ) {
 		createPickerSession,
 		fetchPickerSession,
 		getPickerStatus,
+		setAuthenticated,
 	} = props;
 	const [ cachedSessionId ] = useState( getGooglePhotosPickerCachedSessionId() );
 	const [ isCachedSessionChecked, setIsCachedSessionChecked ] = useState( false );
 	const [ pickerFeatureEnabled, setPickerFeatureEnabled ] = useState( null );
+	const [ authUpgradeRequired, setAuthUpgradeRequired ] = useState( false );
 	const isPickerSessionAccurate = pickerSession !== null && ! ( 'code' in pickerSession );
 	const isSessionExpired =
 		pickerSession?.expireTime && moment( pickerSession.expireTime ).isBefore( new Date() );
 
+	const catchAuthErrors = useCallback(
+		error => {
+			if ( error.code === 'authorization_required' ) {
+				setAuthenticated( false );
+			}
+
+			// If the picker session endpoint returns a 404
+			// the user needs to upgrade their auth
+			if ( error.code === 'rest_not_found' ) {
+				setAuthUpgradeRequired( true );
+			}
+		},
+		[ setAuthenticated, setAuthUpgradeRequired ]
+	);
+
+	useEffect( () => {
+		! isAuthenticated && setAuthUpgradeRequired( false );
+	}, [ isAuthenticated ] );
+
 	useEffect( () => {
 		getPickerStatus().then( feature => {
 			feature && setPickerFeatureEnabled( feature.enabled );
 		} );
 
 		cachedSessionId === null && setIsCachedSessionChecked( true );
-		cachedSessionId &&
-			fetchPickerSession( cachedSessionId ).then( () => setIsCachedSessionChecked( true ) );
-	}, [ getPickerStatus, fetchPickerSession, cachedSessionId ] );
+		fetchPickerSession( cachedSessionId )
```

Comment:
Looks like if I have an OLD connection and not having the picker feature turn on this will still gets fired and result in the upgrade screen too. Should we only fire this if the picker feature is enabled?

Reply:
Yeah, it's true, but it's intentional.
There is a case when you have established a picker connection, and, for example, we do feature revert.

In that case, the `authenticated` flag is true, which is correct, but the connection is not valid, and the only way in this circumstance is to run the session fetch and catch the error.

To be clear, I followed the same approach to checking if the connection is valid, see these lines: https://github.com/Automattic/jetpack/blob/2ec277e1829f672641fea2559d4baff6b1e77ff8/projects/plugins/jetpack/extensions/shared/external-media/sources/with-media.js#L122-L125

Invoked from here:
https://github.com/Automattic/jetpack/blob/2ec277e1829f672641fea2559d4baff6b1e77ff8/projects/plugins/jetpack/extensions/shared/external-media/sources/with-media.js#L146-L181

So, initially, the `authenticated` flag is `true`, and error handling will catch if it requires an authentication process.

Reply:
For the scenario you mentioned I don't have questions, the issue I had is for users with old connection and the picker feature flag off, this means they'll be using the current old flow. And the current implementation will firing the get session check and result in Auth upgrade screen, which they actually have a valid connection.

Reply:
You were right; there was a case when we rendered the AuthUpgrade screen in the wrong place.

It's fixed now, and it is ready for another round of review.

Also, I have updated the cookie domain configuration, so the cookie scope is now correct.